source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -257,7 +282,7 @@ void *ns_dyn_mem_alloc(int16_t alloc_size)
 
 
         if (mem_stat_info_ptr) {
-            alloc_size = (alloc_sector * sizeof(int));
+            alloc_size = ((alloc_sector+2) * sizeof(int));
```

Comment:
Don't think this works - alloc_sector reaches here as 0 if the allocation fails (I think) so this would increase allocation on failure.